### PR TITLE
File exporter

### PR DESCRIPTION
**Pull request by RYCBABAC**
Created file exporter

### CODE REVIEW

File: cards_exporter/cards_pdf_exporter.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from jinja2 import Environment, FileSystemLoader
+from typing import List, NoReturn
+from entities.card import Card
+from cards_exporter.icardexporter import ICardExporter
+from Utils.configuration_keys import ConfigurationKeys
+
+
+class CardPdfExporter(ICardExporter):
+
+    def __init__(self, configuration: dict):
```
Comment (Iftah1) [inline cards_exporter/cards_pdf_exporter.py]: add type of the variables in the key and value, you welcome to ask if you dont know how to :)

File: cards_exporter/cards_pdf_exporter.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import pdfkit
```
Comment (Iftah1) [inline cards_exporter/cards_pdf_exporter.py:1]: no documantion, those function are really hard to understand for someone who doesn't know the libraries

File: cards_exporter/cards_pdf_exporter.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        cards_in_row: int = self._configuration[ConfigurationKeys.NUM_CARDS_IN_ROW_KEY]
+        for i in range(0, len(cards), cards_in_row):
+            yield cards[i: min(i + cards_in_row, len(cards))]
+
+    def render_cards_template(self, cards: List[List[Card]]) -> str:
+        template_file_path: str = self._configuration[ConfigurationKeys.TEMPLATE_FILE_PATH_KEY]
+        template = self._env.get_template(template_file_path)
+        return template.render(rows=cards)
+
+    def write_cards_to_file(self, cards: str, output_path: str):
```
Comment (Iftah1) [inline cards_exporter/cards_pdf_exporter.py]: typing in return value

Comment (Iftah1) [inline cards_exporter/cards_pdf_exporter.py]: how is variable name called cards is str? probably bad naming

File: cards_exporter/cards_pdf_exporter.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+from entities.card import Card
+from cards_exporter.icardexporter import ICardExporter
+from Utils.configuration_keys import ConfigurationKeys
+
+
+class CardPdfExporter(ICardExporter):
+
+    def __init__(self, configuration: dict):
+        self._env = Environment(loader=FileSystemLoader('.'))
+        self._configuration = configuration
```
Comment (Iftah1) [inline cards_exporter/cards_pdf_exporter.py:15]: good job with the private conventions :)

File: cards_exporter/styles/answer-card-title-style.css [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#cards #answer-card h3 {
```
Comment (Iftah1) [inline cards_exporter/styles/answer-card-title-style.css:1]: do i need to go over this as well? or should I trust you?

File: entities/card.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+from dataclasses import dataclass
+from entities.card_type import CardDesignProperties
+
+
+@dataclass
+class Card:
+    content: str
+    type: CardDesignProperties
```
Comment (Iftah1) [inline entities/card.py]: I dont like the type of the card is define with content and design stuff, if you need values for the design stuff, we can define card with:
content: str
design_properties: CardDesignProperties
metadata: CardMetadata

File: entities/card_type.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+from enum import Enum
+from dataclasses import dataclass
+from Utils.configuration_keys import ConfigurationKeys
+
+
+class CardType(Enum):
+    QUESTION = True,
+    ANSWER = False
+
+    def GetDesignPropertiesKey(self) -> str:
```
Comment (Iftah1) [inline entities/card_type.py]: what is going on, lets go over the code in zoom

File: main.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
+import json
```
Comment (Iftah1) [inline main.py]: you said the code isnt for the project right? you can open python file in scratches and it wont upload to git :)

File: main.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        Card("It creates PDF from cards", CardDesignProperties(answer)),
+        Card("Just like the one we need for House of Cards but who really cares?", CardDesignProperties(answer)),
+        Card("And I think it works very well and it's really cool", CardDesignProperties(question)),
+        Card("BTW if you don't think so you can bite me", CardDesignProperties(answer)),
+        Card("So now that it's ready I can finally tell you what I need properly", CardDesignProperties(question)),
+        Card("This is the way. Couldn't do it in any other way", CardDesignProperties(answer)),
+        Card("Are you ready for it?", CardDesignProperties(question)),
+        Card("I'm sure you want to hear it already", CardDesignProperties(answer)),
+        Card("So here it comes", CardDesignProperties(question)),
+        Card("I'M BATMANNNNNNNNNNNNNNNN", CardDesignProperties(answer))
```
Comment (Iftah1) [inline main.py]: No. You are not.

File: cards_exporter/styles/answer-card-title-style.css (hunk 5 above)
Comment (RYCBABAC) [reply cards_exporter/styles/answer-card-title-style.css:1]: As I texted you, I'd love it if you will.  Eventually this is what Matan would see and I want it to look good

File: entities/card.py (hunk 6 above)
Comment (RYCBABAC) [reply entities/card.py]: Totaly right, my bad.
Fixed

File: main.py (hunk 9 above)
Comment (RYCBABAC) [reply main.py]: YES I AM 
I'M THE BATMANNNNNNNNNNNNN

File: main.py (hunk 8 above)
Comment (RYCBABAC) [reply main.py]: What is scratches? Can't find it in `.gitignore`

File: entities/card_type.py (hunk 7 above)
Comment (RYCBABAC) [reply entities/card_type.py]: As said, I'd love it if you could be more specific. 
Anyway this way moved

File: cards_exporter/cards_pdf_exporter.py (hunk 4 above)
Comment (RYCBABAC) [reply cards_exporter/cards_pdf_exporter.py:15]: The only problem here is that it's only a convention!!!!!

File: cards_exporter/cards_pdf_exporter.py (hunk 3 above)
Comment (RYCBABAC) [reply cards_exporter/cards_pdf_exporter.py]: Done

Comment (RYCBABAC) [reply cards_exporter/cards_pdf_exporter.py]: My bad, fixed

File: cards_exporter/cards_pdf_exporter.py (hunk 1 above)
Comment (RYCBABAC) [reply cards_exporter/cards_pdf_exporter.py]: Done (but pleaseeeee let's make a configuration class it looks horrible)

File: entities/card_design_properties.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    sign: str
+    design_id: str
+
+    def __init__(self, json_dict: Dict[str, Any]):
+        self.title = json_dict["title"]
+        self.sign = json_dict["sign"]
+        self.design_id = json_dict["design_id"]
+
+    @staticmethod
+    def GetDesignPropertiesConfigurationKey(card_type: CardType) -> str:
```
Comment (Iftah1) [inline entities/card_design_properties.py]: python conventions!!!!!!

File: entities/card_type.py [hunk 11]
```diff
@@ -0,0 +1,6 @@
+from enum import Enum
+
+
+class CardType(Enum):
```
Comment (Iftah1) [inline entities/card_type.py:4]: change back to answer and question in the enum value

File: cards_exporter/cards_pdf_exporter.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        cards_in_rows = self.split_cards_to_rows(cards)
+        rendered_cards = self.render_cards_template(cards_in_rows)
+        self.write_cards_to_file(rendered_cards, output_path)
+
+    def split_cards_to_rows(self, cards: List[Card]) -> List[List[Card]]:
+        cards_in_row: int = self._configuration[ConfigurationKeys.NUM_CARDS_IN_ROW_KEY]
+        for i in range(0, len(cards), cards_in_row):
+            yield cards[i: min(i + cards_in_row, len(cards))]
+
+    def render_cards_template(self, cards: List[List[Card]]) -> str:
```
Comment (Iftah1) [inline cards_exporter/cards_pdf_exporter.py]: add type for Row of cards

File: entities/card_design_properties.py (hunk 10 above)
Comment (RYCBABAC) [reply entities/card_design_properties.py]: Shut the f*** up

File: entities/card_type.py (hunk 11 above)
Comment (RYCBABAC) [reply entities/card_type.py:4]: What ev'

File: cards_exporter/cards_pdf_exporter.py (hunk 12 above)
Comment (RYCBABAC) [reply cards_exporter/cards_pdf_exporter.py]: Done

**Pull request merged**